sdk/identity/Azure.Identity/src/ClientSecretCredential.cs

```diff
@@ -59,28 +73,40 @@ public ClientSecretCredential(string tenantId, string clientId, string clientSec
         /// <param name="clientSecret">A client secret that was generated for the App Registration used to authenticate the client.</param>
         /// <param name="options">Options that allow to configure the management of the requests sent to the Azure Active Directory service.</param>
         public ClientSecretCredential(string tenantId, string clientId, string clientSecret, TokenCredentialOptions options)
-            : this (tenantId, clientId, clientSecret, CredentialPipeline.GetInstance(options))
+            : this(new MsalConfidentialClientOptions(tenantId: tenantId ?? throw new ArgumentNullException(nameof(tenantId)),
```

Comment:
Shouldn't we do argument null checks in `MsalConfidentialClientOptions` itself?

Reply:
I put these checks here for a couple of reasons. First `MsalConfidentialClientOptions` is internal so I wanted to leave it out of the callstack. Secondly at one point I had some discrepency between the argument names here and in `MsalConfidentialClientOptions` so in order to be able to tie the message to nameof(*) I left the checks here.